### PR TITLE
src: send_patch: Rename `kw mail` feature to `kw send-patch`

### PR DESCRIPTION
**Pull request by lucasqml**
The `kw mail` feature has a name that can be improved to be more descriptive

As said in https://github.com/kworkflow/kworkflow/issues/1027, the new name suggested was send-patch. I think this name summarize better the feature than just `mail`.

This PR consolidates this renaming. It only changes instances of mail to send-patch in file/directory names, function names, and documentation (be it the project's documentation or in comments).

### CODE REVIEW

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Please squash all of these commits into a single one.

File: documentation/conf.py
Comment (rodrigosiqueira) [inline documentation/conf.py:1]: Your commit message looks weird; could you check it?

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi,
This change lgtm. Could you just update it against the latest unstable branch and update this PR?

**Comment by rodrigosiqueira** [comment]
@lucasqml  you can squash all your commits in a single one.

**Review by davidbtadokoro — CHANGES_REQUESTED** [review]
I guys, and thank you for continuing evolving this PR. It may seem like a simple thing, but good names are fundamental!

Given that there a lot of references to the old feature name combined with the fact that the generic name 'mail' doesn't help, I am being a little more careful on this review. Looking at it again, I noticed that we missed an important renaming pertaining to the configurations of the feature. Besides converting the configuration filename from `etc/mail.config` to `etc/send_patch.config`, replace the occurrences of the strings `mail_config` (data struct with configurations at runtime) and `mail.config` with `send_patch_config` and `send_patch.config` in the source files, tests, and docs.

Another important request is the fixing of the commit message body.

**Comment by rodrigosiqueira** [comment]
@lucasqml could you address @davidbtadokoro comments? In particular, update your commit message.
Also, notice that the integration test pointed out a valid issue.

**Comment by davidbtadokoro** [comment]
Hey @lucasqml and @bolgheroni. Any updates here?

To fix the integration tests fail you need to do the following:

```bash
--- a/tests/integration/config_test.sh
+++ b/tests/integration/config_test.sh
@@ -61,9 +61,9 @@ function test_deploy_config()
   local_config_test_helper 'deploy'
 }
 
-function test_mail_config()
+function test_send_patch_config()
 {
-  local_config_test_helper 'mail'
+  local_config_test_helper 'send_patch'
 }
 
 function test_notification_config()
```

I've noticed a crucial update in `src/mail.sh`. As we've renamed the feature as a whole and its config file and references (both variable names and functions) we need to update the call to `load_mail_config` to `load_send_patch_config`. The way to do it is below:

```bash
--- a/src/send_patch.sh
+++ b/src/send_patch.sh
@@ -1151,4 +1151,4 @@ function send_patch_help()
     '  send-patch --verbose - Show a detailed output'
 }
 
-load_mail_config
+load_send_patch_config
```

Last (but certainly not least) don't forget to write a commit message body to give this important change a context! I know it may seem straightforward, but try to explain the motivation in changing the feature's name (because it was too generic) and how `send-patch` is better. 

**Obs.:** @rodrigosiqueira, @melissawen, @aquilamacedo, @JGBSouza, @MarceloSpessoto, and anyone else, feel free to weigh-in on the name `send-patch` and/or suggest other name candidates.

**Comment by lucasqml** [comment]
Hi @rodrigosiqueira @davidbtadokoro !

Sorry for the delay in responding, but during the time I was offline, I had to undergo surgery and had some complications during recovery.

However, I was able to take a look at the reviews now and corrected the changes! If you could take another look, please!

**Comment by rodrigosiqueira** [comment]
Thanks for working on this PR. Change merged into the unstable branch.

**Pull request closed without merging**